### DISCUSSION TITLE
Using .SELECT with DISTINCT and List<string>.

### DISCUSSION THREAD

**Discussion by DynamicCommander** [maxtoroq/DbExtensions#58] [Q&A] [answered]
Hello again.

So what I'm trying to do is pass a list of column names that I want to use DISTINCT on. However, I can't seem to figure out how the param[] parameter works. When I call for example .SELECT("DISTINCT ", myList.ToArray()); The resulting query does not contain the array of strings that I passed into it. 

Also I am having trouble understanding how to go from '@p0' to 'Wisconsin' which is one of the parameter values (Not using a list in this case).

Thanks again!

**Comment by maxtoroq** [comment, marked as answer]
The *params* parameter is for parameter values only, not for column names. Do a simple string join:

```csharp
query.SELECT("DISTINCT " + string.Join(", ", myList));
```

**Comment by DynamicCommander** [comment]
Ah okay, that seems like a good solution to that. I'm new to SQL so trying to learn SQL and DbExtensions at the same time is quite the task. 

Thanks for your help!
